File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/dominio/pagamento/MultaDoPagamento.java
/*
 * Decompiled with CFR 0.152.
 * 
 * Could not load the following classes:
 *  javax.persistence.Column
 *  javax.persistence.Entity
 *  javax.persistence.EnumType
 *  javax.persistence.Enumerated
 *  javax.persistence.FetchType
 *  javax.persistence.GeneratedValue
 *  javax.persistence.GenerationType
 *  javax.persistence.Id
 *  javax.persistence.JoinColumn
 *  javax.persistence.ManyToOne
 *  javax.persistence.OneToOne
 *  javax.persistence.SequenceGenerator
 *  javax.persistence.Table
 *  org.hibernate.annotations.Parameter
 *  org.hibernate.annotations.Type
 *  org.hibernate.validator.NotNull
 *  org.jboss.seam.annotations.Name
 */
package br.jus.trt8.pjecalc.negocio.dominio.pagamento;

import br.jus.trt8.pjecalc.negocio.comum.EntidadeAgregada;
import br.jus.trt8.pjecalc.negocio.constantes.TipoVinculoDeMultaDoPagamentoEnum;
import br.jus.trt8.pjecalc.negocio.dominio.calculo.multa.Multa;
import br.jus.trt8.pjecalc.negocio.dominio.pagamento.Pagamento;
import java.math.BigDecimal;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;
import javax.persistence.SequenceGenerator;
import javax.persistence.Table;
import org.hibernate.annotations.Parameter;
import org.hibernate.annotations.Type;
import org.hibernate.validator.NotNull;
import org.jboss.seam.annotations.Name;

@Entity
@Table(name="TBVINCULOPAGAMENTOMULTA")
@SequenceGenerator(name="SQVINCULOPAGAMENTOMULTA", sequenceName="SQVINCULOPAGAMENTOMULTA", allocationSize=1)
@Name(value="multaDoPagamento")
public class MultaDoPagamento
extends EntidadeAgregada {
    private static final long serialVersionUID = 4466529296239033805L;
    @Id
    @GeneratedValue(strategy=GenerationType.SEQUENCE, generator="SQVINCULOPAGAMENTOMULTA")
    @Column(name="IIDVINCULOPAGAMENTOMULTA")
    private final Long id = null;
    @ManyToOne(fetch=FetchType.LAZY)
    @JoinColumn(name="IIDMULTA")
    private Multa multa;
    @OneToOne
    @JoinColumn(name="IIDPAGAMENTO")
    private Pagamento pagamento;
    @Column(name="STPVINCULO", columnDefinition="VARCHAR2(1)")
    @NotNull
    @Type(type="br.jus.trt8.pjecalc.base.comum.EnumUserType", parameters={@Parameter(name="enum", value="TipoVinculoDeMultaDoPagamentoEnum")})
    private TipoVinculoDeMultaDoPagamentoEnum tipoVinculo = TipoVinculoDeMultaDoPagamentoEnum.DEBITOSRECLAMANTE;
    @Enumerated(value=EnumType.STRING)
    @Column(name="SFLAPURARMULTA", nullable=false, columnDefinition="CHAR(1)")
    @Type(type="br.jus.trt8.pjecalc.base.comum.BooleanUserType")
    @NotNull
    private Boolean apurarMulta = true;
    @Column(name="MVLMULTA", precision=12, scale=2)
    private BigDecimal valorMulta;

    public MultaDoPagamento() {
    }

    public MultaDoPagamento(Multa multa, Pagamento pagamento, BigDecimal valorMulta, Boolean apurarMulta, TipoVinculoDeMultaDoPagamentoEnum tipoVinculo) {
        this();
        this.multa = multa;
        this.pagamento = pagamento;
        this.valorMulta = valorMulta;
        this.apurarMulta = apurarMulta;
        this.tipoVinculo = tipoVinculo;
    }

    public MultaDoPagamento(MultaDoPagamento pagamentoParaClonar) {
        this(pagamentoParaClonar.getMulta(), pagamentoParaClonar.getPagamento(), pagamentoParaClonar.getValorMulta(), pagamentoParaClonar.getApurarMulta(), pagamentoParaClonar.getTipoVinculo());
    }

    public Long getId() {
        return this.id;
    }

    @Override
    public Object obterChavePrimaria() {
        return this.getId();
    }

    public Multa getMulta() {
        return this.multa;
    }

    public void setMulta(Multa multa) {
        this.multa = multa;
    }

    public Pagamento getPagamento() {
        return this.pagamento;
    }

    public void setPagamento(Pagamento pagamento) {
        this.pagamento = pagamento;
    }

    public TipoVinculoDeMultaDoPagamentoEnum getTipoVinculo() {
        return this.tipoVinculo;
    }

    public void setTipoVinculo(TipoVinculoDeMultaDoPagamentoEnum tipoVinculo) {
        this.tipoVinculo = tipoVinculo;
    }

    public BigDecimal getValorMulta() {
        return this.valorMulta;
    }

    public void setValorMulta(BigDecimal valorMulta) {
        this.valorMulta = valorMulta;
    }

    public Boolean getApurarMulta() {
        return this.apurarMulta;
    }

    public void setApurarMulta(Boolean apurarMulta) {
        this.apurarMulta = apurarMulta;
    }

    @Override
    public int hashCode() {
        int prime = 31;
        int result = 1;
        result = 31 * result + (this.id == null ? 0 : this.id.hashCode());
        return result;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (this.getClass() != obj.getClass()) {
            return false;
        }
        MultaDoPagamento other = (MultaDoPagamento)obj;
        if (this.pagamento == null ? other.pagamento != null : !this.pagamento.equals(other.pagamento)) {
            return false;
        }
        if (this.id == null ? other.id != null : !this.id.equals(other.id)) {
            return false;
        }
        if (this.tipoVinculo != other.tipoVinculo) {
            return false;
        }
        return !(this.multa == null ? other.multa != null : !this.multa.equals(other.multa));
    }
}

File: src/lib/pjecalc/core/dominio/pagamento/multa-do-pagamento.ts
/**
 * PJe-Calc v2.15.1 — MultaDoPagamento
 * Porte 1:1 de: br.jus.trt8.pjecalc.negocio.dominio.pagamento.MultaDoPagamento
 *
 * Ref Java: pjecalc-fonte/.../pagamento/MultaDoPagamento.java
 *
 * Vínculo Multa ↔ Pagamento (3 tipos de vínculo) com valor + apurarMulta.
 */
import type Decimal from 'decimal.js';
import { TipoVinculoDeMultaDoPagamentoEnum } from '../../constantes/enums';
import type { Multa } from '../calculo/multa/multa';
import type { Pagamento } from './pagamento';

export class MultaDoPagamento {
  private id: number | null = null;
  private multa: Multa | null = null;
  private pagamento: Pagamento | null = null;
  private tipoVinculo: TipoVinculoDeMultaDoPagamentoEnum = TipoVinculoDeMultaDoPagamentoEnum.DEBITOSRECLAMANTE;
  private apurarMulta: boolean = true;
  private valorMulta: Decimal | null = null;

  constructor(args?: MultaDoPagamento | {
    multa: Multa | null;
    pagamento: Pagamento | null;
    valorMulta: Decimal | null;
    apurarMulta: boolean;
    tipoVinculo: TipoVinculoDeMultaDoPagamentoEnum;
  }) {
    if (!args) return;
    if (args instanceof MultaDoPagamento) {
      this.multa = args.getMulta();
      this.pagamento = args.getPagamento();
      this.valorMulta = args.getValorMulta();
      this.apurarMulta = args.getApurarMulta();
      this.tipoVinculo = args.getTipoVinculo();
    } else {
      this.multa = args.multa;
      this.pagamento = args.pagamento;
      this.valorMulta = args.valorMulta;
      this.apurarMulta = args.apurarMulta;
      this.tipoVinculo = args.tipoVinculo;
    }
  }

  getId(): number | null { return this.id; }
  setId(v: number): void { this.id = v; }

  getMulta(): Multa | null { return this.multa; }
  setMulta(m: Multa | null): void { this.multa = m; }

  getPagamento(): Pagamento | null { return this.pagamento; }
  setPagamento(p: Pagamento | null): void { this.pagamento = p; }

  getTipoVinculo(): TipoVinculoDeMultaDoPagamentoEnum { return this.tipoVinculo; }
  setTipoVinculo(v: TipoVinculoDeMultaDoPagamentoEnum): void { this.tipoVinculo = v; }

  getValorMulta(): Decimal | null { return this.valorMulta; }
  setValorMulta(v: Decimal | null): void { this.valorMulta = v; }

  getApurarMulta(): boolean { return this.apurarMulta; }
  setApurarMulta(v: boolean): void { this.apurarMulta = v; }
}
